### PR TITLE
terms: add 'immutable' to DLT definition

### PR DESCRIPTION
**Pull request by rhiaro**
Suggested by @iherman in #401. Obviously not an expert, my understanding is a DLT is kinda shoulda supposed to be immutable but under certain situations this might not technically be the case; but for the purposes of this definition stating it as Ivan suggests I think is probably right. Look, now this comment took longer to write than making the change itself. Just close the PR if it's not appropriate to add "immutable" anyway.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/pull/450.html" title="Last updated on Nov 11, 2020, 2:19 PM UTC (dba8371)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/450/b426d48...dba8371.html" title="Last updated on Nov 11, 2020, 2:19 PM UTC (dba8371)">Diff</a>

### CODE REVIEW

**Review by iherman — APPROVED** [review]
I am obviously o.k. with it, but it is not my voice that counts...

**Review by peacekeeper — APPROVED** [review]

**Comment by jandrieu** [comment]
-1
Even DLTs are not "immutable". This is an unfortunate term that has caused considerable hype and grief.

Perhaps "distributed" addresses your goal better:
> to maintain an distributed ledger in which each transaction is cryptographically

**Comment by iherman** [comment]
@jandrieu I am obviously not an expert, and I am not sure what the optimal term is. However, in our world isn't the fact that a node put on the DLT is not supposed to change an essential feature?

**Comment by msporny** [comment]
While there are corner cases where what @jandrieu says is true, DLTs are viewed to be statistically immutable in the general case. Could you elaborate more on your concern, @jandrieu? I don't think your rewording addresses @iherman's initial issue raised.

**Comment by jandrieu** [comment]
Hmmm.

I didn't realize my suggestion would end up defining DLTs in terms of distributed ledgers, which definitely doesn't work.

So, I would suggest just not adding "immutable".

My concern is both technical and cultural. 

DLTs are not, in fact, immutable. They fork all the time. Sometimes as a result of intentional manipulation. At best, what is established is a probabilistic conviction that it is economically infeasible for attackers to make more money subverting the chain for a double spend than they would make by mining. 

In many cases, this economic feasibility argument fails--either because there are not enough interested computation to keep the system honest or because the goals of the attacker are not economic. So, only the best DLTs actually achieve ANY kind of immutability--and it still based on probabilistic economics rather than factual immutability. Many, perhaps most DLTs are constantly at risk of this type of attack, which is why foundational layer 1 ledgers are deemed so valuable: the feedback loop of value increases the security profile so valuable chains get more valuable. So, technically, it is not only incorrect to say a DLT is immutable, it is at best aspirational hopes and at worst willful disinformation for those DLTs that have not achieved the right economic balance. 

Conditional immutability is a goal, not a fact, of DLTs.

Culturally, the problem is that "immutable" was the "sizzle" in many businesses' pitch for ICOs and blockchain fairy dust. As such, it has acquired a patina of huksterism both within the technical communities working on blockchain tech as well as the investors and customers that bought into the hype of "immutability". 

So, although well-intended, I think it diverts more than it adds to the conversation.

**Comment by msporny** [comment]
> So, only the best DLTs actually achieve ANY kind of immutability--and it still based on probabilistic economics rather than factual immutability.

Ah, so here is the point of contention -- the above is not true for Paxos, RAFT, Veres One's Continuity consensus algorithm as well as the Hashgraph consensus algorithm. There are consensus algorithms with provable finality. Bitcoin and Ethereum don't have it, the best they can do is statistical, but there are a whole class of consensus algorithms that do achieve finality. 

That's what I was trying to point out... your statement doesn't hold for the entire class of technologies.

**Comment by jandrieu** [comment]
Indeed. There are exceptions. Which means that arguably, *some* DLTs are immutable. Some aren't.

**Comment by msporny** [comment]
> Indeed. There are exceptions. Which means that arguably, _some_ DLTs are immutable. Some aren't.

Ah, ok, we're on the same page then... you said this before:

> DLTs are not, in fact, immutable.

... which is what raised my eyebrow. 

Eyebrow is now lowered again.

**Comment by msporny** [comment]
Soooooo... this means that this PR should be closed w/o being merged?

**Comment by iherman** [comment]
> Soooooo... this means that this PR should be closed w/o being merged?

Well... I go back to my original problem that led to this issue here. I am a layperson and read this in the terminology of our spec:

> **distributed ledger (DLT)**
> A distributed database in which the various nodes use a consensus protocol to maintain a shared ledger in which each transaction is cryptographically signed and chained to the previous transaction.

For that layperson reader I believe one of the main point of using a DLT is that if I create something that is put on a DLT, that something cannot change any more. It is cast in concrete, marble, titanium, whatever, but I (or somebody) else cannot go and change it. The "which each transaction is cryptographically signed and chained to the previous transaction" is a technical mumbo-jumbo that does not help the reader to grasp the essence of it.

I frankly do not care whether the term "immutable" is the right technical one here. But I still believe part of an essential information gets lost.

**Comment by msporny** [comment]
@iherman wrote:
> I frankly do not care whether the term "immutable" is the right technical one here. But I still believe part of an essential information gets lost.

Got it -- that's a reasonable expectation -- @jandrieu thoughts and concrete language that address @iherman's concerns?

**Comment by jandrieu** [comment]
It's not our job to explain why DLTs are useful. We're just trying to define them in a way that is accurate. "immutable" is not accurate.

Here's the existing language:
> distributed ledger (DLT)
> A distributed database in which the various nodes use a consensus protocol to maintain a shared ledger in which each transaction is cryptographically signed and chained to the previous transaction.

This is accurate and complete.

I don't think we want to put marketing buzzwords like "immutable" in a terminological definition in a specification.

**Comment by msporny** [comment]
What about this, @jandrieu, @iherman:

> distributed ledger (DLT)
> A non-centralized system for recording events. These systems often achieve sufficient trust among those that use the system to make decisions based on the information in the system. In more technical terms, solutions often consist of distributed databases in which the various database nodes in the system use a consensus protocol in which each transaction is cryptographically signed and chained to the previous transaction, often making the history of the ledger effectively immutable.

**Comment by iherman** [comment]
Works for me, @msporny

**Comment by peacekeeper** [comment]
"**append-only** ledger"?

Just an idea, no strong opinion here...

**Comment by jandrieu** [comment]
+1 @msporny The basics of that are great. An iteration:
> distributed ledger (DLT)
> A non-centralized system for recording events. These systems establish sufficient confidence for participating users to rely upon the data recorded by others to make operational decisions. They typically use distributed databases where different nodes use a consensus protocol to confirm the ordering of cryptographically signed transactions. The linking of digitally signed transactions over time often makes the history of the ledger effectively immutable.

**Review by OR13 — APPROVED** [review]

**Review by kdenhartog — APPROVED** [review]

**Comment by rhiaro** [comment]
I've updated the definition to use @jandrieu's iteration on @msporny's text https://github.com/w3c/did-core/pull/450#issuecomment-722515284

Sorry to those who recently approved. Not sure if you were approving the original change or Joe's update. I'll re-request review just in case.

File: terms.html [hunk 1]
```diff
@@ -213,11 +213,13 @@
   <dt><dfn data-lt="distributed ledger technology|DLT">distributed ledger</dfn> (DLT)</dt>
 
   <dd>
-A <a href="https://en.wikipedia.org/wiki/Distributed_database">distributed database</a>
-in which the various nodes use a
-<a href="https://en.wikipedia.org/wiki/Consensus_(computer_science)">consensus protocol</a>
-to maintain a shared ledger in which each transaction is cryptographically
-signed and chained to the previous transaction.
+A non-centralized system for recording events. These systems establish
+sufficient confidence for participating users to rely upon the data recorded
```
Comment (msporny) [inline terms.html:217]:
```suggestion
sufficient confidence for participants to rely upon the data recorded
```

**Review by msporny — APPROVED** [review]
Minor editorial nit, otherwise LGTM.

**Review by peacekeeper — APPROVED** [review]

**Review by kdenhartog — APPROVED** [review]
Happy with the new definition as well!

**Review by brentzundel — APPROVED** [review]

**Comment by msporny** [comment]
Editorial, multiple positive reviews, changes requested and made, no objections, merging.

**Pull request merged**